### ISSUE TITLE
Bug: `no-double-punctuation` reports punctuation after gemoji and HTML entities

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#758]
### Environment

ESLint version: 9.39.5
@eslint/markdown version: 8.0.3
eslint-markdown version: 0.17.0
Node version: 24.14.0
npm version: 11.9.0
Operating System: macOS 26.5

### Which language are you using?

commonmark & gfm

### What did you do?

<details>
<summary>Configuration</summary>

```js
import { defineConfig } from 'eslint/config';
import markdown from '@eslint/markdown';
import md from 'eslint-markdown';

export default defineConfig([
  {
    files: ['**/*.md'],
    plugins: {
      markdown,
      md,
    },
    language: 'markdown/commonmark',
    rules: {
      'md/no-double-punctuation': 'error',
    },
  },
]);
```

</details>

```md
This is :smile:.
Copyright &copy;.
```

Run the following command:

```sh
npm run lint
```

The same behavior occurs with both CommonMark and GFM.
The same behavior occurs with both CommonMark and GFM.

### What did you expect to happen?

No errors should be reported.

The colon is part of the gemoji, and the semicolon is part of the HTML entity. They should not be treated as consecutive punctuation with the period that follows.

### What actually happened?

The rule reports `:.` and `;.` as double punctuation:

<img width="468" height="219" alt="Image" src="https://github.com/user-attachments/assets/2c20c893-ddf7-4139-a77b-5f2b43ebdc4e" />

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-s1olntc4?file=eslint.config.js,commonmark.md

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] No - I did not use AI to generate this issue report.
- [x] Yes - minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes - substantive assistance (AI-generated non-trivial portions of code, tests, or documentation).

**Comment by lumirlumir** [comment]
Thanks for the issue. A PR is welcome!

**Comment by tooth-is-silver** [comment]
@lumirlumir I’ll also move `trailingHtmlEntityRegex` from `no-trailing-heading-punctuation` to the shared constants and add tests for it as part of this work.
